docs/lib/datastore/fragments/ios/auth-model.md

```diff
@@ -0,0 +1,170 @@
+
+1. User's sign in state is known to be signed out, and the user needs to sign in, prior to making a call to `DataStore.save()`. 
+
+```swift
+Amplify.Auth.signIn {
+    switch $0 {
+    case .success(let signInResult):
+        if signInResult.isSignedIn {
+            // Amplify.DataStore.save
+        }
+    case .failure(let error):
+        print("Sign in error \(error)")
+    }
+}
+```
+
+2. User's sign in state is unknown (for example, app was terminated or transitioned from background to foreground) and you need to check prior to making a call to `Datastore.save()`. 
+
+```swift
+_ = Amplify.Auth.fetchAuthSession {
+    switch $0 {
+    case .success(let authSession):
+        if authSession.isSignedIn {
+            // Amplify.DataStore.save
+        }
+    case .failure(let error):
+        print("failed to get auth session", error)
+    }
+}
+```
+
+3. During the lifecycle of the app, listen to Auth events to show different views based on the user's state. See [Auth Events](~/lib/auth/auth-events.md) for more details.
+
+4. If the user state becomes signed out during the operation's sync to cloud, you can know exactly which operation failed when the error handler is called. By default, the error will be logged by `Amplify.Logging.error`
+
+```swift
+let configuration = DataStoreConfiguration.custom(errorHandler: { error in
+    Amplify.Logging.error(error: error)
+
+    if let dataStoreError = error as? DataStoreError,
+        case let .api(amplifyError, mutationEventOptional) = dataStoreError,
+        let actualAPIError = amplifyError as? APIError,
+        case let .operationError(_, _, underlyingError) = actualAPIError,
+        let authError = underlyingError as? AuthError,
+        let mutationEvent = mutationEventOptional {
+
+        if case .signedOut = authError {
+            let modelName = mutationEvent.modelName
+            let id = mutationEvent.modelId
+            let mutationType = mutationEvent.mutationType
+            print("The \(mutationType) operation for \(modelName) with \(id) failed due to user signed out.")
+        }
+    }
+})
+```
+
+For more information on related topics:
+
+- [Auth.SignIn](~/lib/auth/signin.md)
+- [Auth.fetchAuthSession](~/lib/auth/getting-started.md#check-the-current-auth-session)
+- [Auth Events](~/lib/auth/auth-events.md)
+- [DataStoreConfiguration handlers](~/lib/datastore/conflict.md)
+
+## Model owner
+
+To check who the owner of the model is, call `DataStore.query` and check the user's username against the owner field. 
+```swift
+if let user = Amplify.Auth.getCurrentUser() {
```

Comment:
Maybe update the variable name to `currentUser`

if let currentUser = Amplify.Auth.getCurrentUser() {
}